torch/testing/_internal/common_utils.py

```diff
@@ -61,6 +61,10 @@
 from statistics import mean
 import functools
 from .composite_compliance import no_dispatch
+from torch.testing._internal.common_dtype import get_all_dtypes
+
+# The implementation should be moved here as soon as the deprecation period is over.
+from torch.testing._legacy import get_all_device_types  # noqa: F401
```

Comment:
@mruberry  how come this doesn't include "meta" as a device type? Is there a better function I should use instead of this legacy function?

Reply:
Yes! You can acquire the available test base classes here:

https://github.com/pytorch/pytorch/blob/b372be4211d40be92ffbb025cac0038ee3445141/torch/testing/_internal/common_device_type.py#L514

and then query each of them for their device type. This allows for programmatically added test bases, like PyTorch/XLA, to be acquired when they're available.